src/tools/compiletest/src/runtest.rs

```diff
@@ -2979,19 +2981,42 @@ impl<'test> TestCx<'test> {
             .with_extra_extension(mode)
             .with_extra_extension(kind);
 
-        match File::create(&output_file).and_then(|mut f| f.write_all(actual.as_bytes())) {
-            Ok(()) => {}
-            Err(e) => self.fatal(&format!(
-                "failed to write {} to `{}`: {}",
-                kind,
-                output_file.display(),
-                e
-            )),
+        let mut files = vec![output_file];
+        if self.config.bless {
+            files.push(self.expected_output_path(kind));
```

Comment:
This doesn't seem to work for `nll` files.  For an example, copy E0508.rs to `src/test/ui` and try to bless it.  You should end up with these three files:
```
E0508.ast.nll.stderr
E0508.ast.stderr
E0508.mir.stderr
```

I think the issue is that `expected_output_path` doesn't include mode if the file doesn't already exist.  I think the solution is to call the global `expected_output_path` (not the `self` version).